spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TempDirSharedInterceptor.java

```diff
@@ -0,0 +1,28 @@
+package org.spockframework.runtime.extension.builtin;
+
+import org.spockframework.runtime.extension.IMethodInvocation;
+import org.spockframework.runtime.model.SpecInfo;
+
+import java.util.function.BiConsumer;
+
+/**
+ * @author dqyuan
+ * @since 2.0
+ */
+public class TempDirSharedInterceptor extends TempDirBaseInterceptor {
+
+  TempDirSharedInterceptor(Class<?> fieldType, BiConsumer<Object, Object> fieldSetter, String parentDir, boolean reserveAfterTest) {
+    super(fieldType, fieldSetter, parentDir, reserveAfterTest);
+  }
+
+  @Override
+  void install(SpecInfo specInfo) {
+    specInfo.addInterceptor(this);
+  }
+
+  @Override
+  public void interceptSpecExecution(IMethodInvocation invocation) throws Throwable {
```

Comment:
Confused with the meaning of `you can simply change the signature of invoke to implement the interface and do not need forwarder methods in the subclasses. `. Can you give a example?  Sub class at least need to tell base interceptor where to intercepor, so I think it will not simplify code too much...

Reply:
Well, it makes the code easier readable as you don't need methods to forward to the one method doing the work and you also save two unnecessary calls on the stack at runtime. The `AbstractMethodInterceptor` is handy if you want to attach the same interceptor class to multiple interception points and do different things for different interception points. But if you want to attach the same class to mutliple interception points or even just one interception point and want to always do the same thing as in this case, there is no need to extend the `AbstractMethodInterceptor`.

What I meant is, delete `TempDirSharedInterceptor#interceptSpecExecution` and `TempDirIterationInterceptor#interceptIterationExecution`, change `extends AbstractMethodInterceptor` to `implements IMethodInterceptor` and `protected void invoke(IMethodInvocation invocation) throws Throwable {` to `@Override\npublic void intercept(IMethodInvocation invocation) throws Throwable {`.

Yes, you would of course still need the `install` method in the subclasses, or you could alternatively do the three lines of attaching directly in the extension and remove the subclasses completely. :-)

Reply:
Got it